supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/MemoryDeclarationValidation.scala

```diff
@@ -0,0 +1,115 @@
+package cromwell.backend.impl.sfs.config
+
+import cromwell.backend.MemorySize
+import cromwell.backend.impl.sfs.config.ConfigConstants._
+import cromwell.backend.validation._
+import wdl4s.expression.NoFunctions
+import wdl4s.parser.MemoryUnit
+import wdl4s.types._
+import wdl4s.values.{WdlFloat, WdlInteger, WdlOptionalValue, WdlValue}
+import wdl4s.{Declaration, NoLookup, WdlExpression}
+
+/**
+  * Maps declarations of memory in WDL runtime attributes to the commands used to submit.
+  *
+  * The wdl runtime attributes for memory specified as strings such as:
+  *
+  * {{{
+  *   runtime {
+  *     memory: "500 MB"
+  *   }
+  * }}}
+  *
+  * However, the backend configuration only supports specifying memory in amounts in Float or Int. To specify the unit
```

Comment:
Thanks for the very detailed explanation ! 
I think strong types are better anyway and we should move to that direction more than the implicit conversion one.

Another idea would be something like

```
Memory myMemory = memory(500, "MB")
```

Anyway that's far from being required here I was just curious of the why :)